src/compiler/checker.ts

```diff
@@ -8570,8 +8701,29 @@ namespace ts {
             return result;
         }
 
+        function getConditionalTypeInstantiation(type: ConditionalType, mapper: TypeMapper): Type {
+            const target = type.target || type;
+            const combinedMapper = type.mapper ? combineTypeMappers(type.mapper, mapper) : mapper;
+            // Check if we have a conditional type of the form T extends U ? X : Y, where T is a type parameter.

```

Comment:
It would be easier to read as:
"A conditional type with a type parameter for its check type is distributive over unions. When T is instantiated to ..."

The "of the form T extends U ? X : Y" is a red herring, because it describes *every* conditional type.